.travis.yml

```diff
@@ -12,30 +12,45 @@ services:
 addons:
   hostname: localhost
 
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+  - rm -f  $HOME/.cache/pip/log/debug.log
+
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+    - $HOME/.cache/pip
+
 before_install:
   - docker --version
   # install docker-compose
   - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
   - chmod +x docker-compose
   - sudo mv docker-compose /usr/local/bin
-
-before_script:
-  - sudo /etc/init.d/postgresql stop
   - pip install --user codecov
   - curl http://www.jpm4j.org/install/script | sudo sh
   - sudo jpm install com.codacy:codacy-coverage-reporter:assembly
 
+install:
+  - ./gradlew compileJava compileTestJava assemble
```

Comment:
default `install` step is calling `gradlew assemble`, that leads to `compileJava` but not `compileTestJava`. see https://docs.travis-ci.com/user/languages/java/#Projects-Using-Gradle

adding `compileTestJava` to the `install` phase ensures, that the complication of test resources is done before running the tests themselves and, hopefully, improves performance of the later matrix script executions. Though I do not understand completely how much of the compilation results is shared between matrix jobs.